hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/common/model/HoodieSparkRecord.java

```diff
@@ -237,14 +237,24 @@ public HoodieRecord truncateRecordKey(Schema recordSchema, Properties props, Str
   }
 
   @Override
-  public boolean isDelete(Schema recordSchema, Properties props) throws IOException {
+  public boolean isDelete(Schema recordSchema, Properties props) {
     if (null == data) {
       return true;
     }
-    if (recordSchema.getField(HoodieRecord.HOODIE_IS_DELETED_FIELD) == null) {
+
+    // Use metadata filed to decide.
+    Schema.Field operationField = recordSchema.getField(OPERATION_METADATA_FIELD);
+    if (operationField != null && "D".equals(data.get(operationField.pos(), StringType))) {
```

Comment:
Maybe we can add a utility method `HoodieOperation.isDelete(String opName)` instead.

Reply:
@danny0405 , I added the util function in HoodieOperation.